src/trs/mod.rs

```diff
@@ -7,6 +7,42 @@
 //!      "Wikipedia - Hindley-Milner Type System"
 //! [1]: https://en.wikipedia.org/wiki/Rewriting#Term_rewriting_systems
 //!      "Wikipedia - Term Rewriting Systems"
+//!
+//! # Example
+//!
+//! ```
+//! # #[macro_use] extern crate polytype;
+//! # extern crate programinduction;
+//! # extern crate term_rewriting;
+//! # use programinduction::trs::{TRS, Lexicon};
+//! # use term_rewriting::{Signature, parse_rule};
+//! # fn main() {
+//! let mut sig = Signature::default();
+//!
+//! let mut ops = vec![];
+//! sig.new_op(2, Some("PLUS".to_string()));
+//! ops.push(ptp![@arrow[tp!(int), tp!(int), tp!(int)]]);
+//! sig.new_op(1, Some("SUCC".to_string()));
+//! ops.push(ptp![@arrow[tp!(int), tp!(int)]]);
+//! sig.new_op(0, Some("ZERO".to_string()));
+//! ops.push(ptp![int]);
+//!
+//! let rules = vec![
+//!     parse_rule(&mut sig, "PLUS(x_ ZERO) = x_").expect("parsed rule"),
+//!     parse_rule(&mut sig, "PLUS(x_ SUCC(y_)) = SUCC(PLUS(x_ y_))").expect("parsed rule"),
+//! ];
+//!
+//! let vars = vec![
+//!     ptp![int],
+//!     ptp![int],
+//!     ptp![int],
+//! ];
+//!
+//! let lexicon = Lexicon::from_signature(sig, ops, vars, vec![], false);
+//!
+//! let trs = TRS::new(&lexicon, rules);
```

Comment:
It'd be nice to do something with this TRS using GP, but that can wait for later.